app/validators/metadata_presenter/autocomplete_validator.rb

```diff
@@ -0,0 +1,21 @@
+module MetadataPresenter
+  class AutocompleteValidator < BaseValidator
+    attr_reader :autocomplete_items
+
+    def initialize(page_answers:, component:, autocomplete_items:)
+      super(page_answers: page_answers, component: component)
+
+      @autocomplete_items = autocomplete_items
+    end
+
+    def invalid_answer?
+      return if autocomplete_item_list.blank?
+
+      autocomplete_item_list.exclude?(JSON.parse(user_answer))
+    end
+
+    def autocomplete_item_list
+      autocomplete_items[component.uuid] || []
```

Comment:
you can memoize this. also since you now have the guard clause there is no need to set the `[]` as an alternative as the guard clause will return if it is `nil`.